mesonbuild/dependencies/misc.py

```diff
@@ -810,90 +811,80 @@ def get_pkgconfig_variable(self, variable_name, kwargs):
 class PcapDependency(ExternalDependency):
     def __init__(self, environment, kwargs):
         super().__init__('pcap', environment, None, kwargs)
-        kwargs['required'] = False
-        if DependencyMethods.PKGCONFIG in self.methods:
+
+    @classmethod
+    def _factory(cls, environment, kwargs):
+        methods = cls._process_method_kw(kwargs)
+        if DependencyMethods.PKGCONFIG in methods:
             try:
                 pcdep = PkgConfigDependency('pcap', environment, kwargs)
                 if pcdep.found():
-                    self.type_name = 'pkgconfig'
-                    self.is_found = True
-                    self.compile_args = pcdep.get_compile_args()
-                    self.link_args = pcdep.get_link_args()
-                    self.version = pcdep.get_version()
-                    self.pcdep = pcdep
-                    return
+                    return pcdep
             except Exception as e:
                 mlog.debug('Pcap not found via pkgconfig. Trying next, error was:', str(e))
-        if DependencyMethods.CONFIG_TOOL in self.methods:
+        if DependencyMethods.CONFIG_TOOL in methods:
             try:
                 ctdep = ConfigToolDependency.factory(
                     'pcap', environment, None, kwargs, ['pcap-config'], 'pcap-config')
                 if ctdep.found():
-                    self.config = ctdep.config
-                    self.type_name = 'config-tool'
-                    self.compile_args = ctdep.get_config_value(['--cflags'], 'compile_args')
-                    self.link_args = ctdep.get_config_value(['--libs'], 'link_args')
-                    self.version = self.get_pcap_lib_version()
-                    self.is_found = True
-                    return
+                    ctdep.compile_args = ctdep.get_config_value(['--cflags'], 'compile_args')
+                    ctdep.link_args = ctdep.get_config_value(['--libs'], 'link_args')
+                    ctdep.version = cls.get_pcap_lib_version(ctdep)
+                    return ctdep
             except Exception as e:
                 mlog.debug('Pcap not found via pcap-config. Trying next, error was:', str(e))
 
-    def get_methods(self):
+        return PcapDependency(environment, kwargs)
+
+    @staticmethod
+    def get_methods():
         if mesonlib.is_osx():
             return [DependencyMethods.PKGCONFIG, DependencyMethods.CONFIG_TOOL, DependencyMethods.EXTRAFRAMEWORK]
         else:
             return [DependencyMethods.PKGCONFIG, DependencyMethods.CONFIG_TOOL]
 
-    def get_pcap_lib_version(self):
-        return self.compiler.get_return_value('pcap_lib_version', 'string',
-                                              '#include <pcap.h>', self.env, [], [self])
+    @staticmethod
+    def get_pcap_lib_version(ctdep):
+        return ctdep.compiler.get_return_value('pcap_lib_version', 'string',
+                                              '#include <pcap.h>', ctdep.env, [], [ctdep])
```

Comment:
__[Flake8]__

[E128] continuation line under-indented for visual indent


<!-- Comment by [SideCI](https://sideci.com) -->